### PR TITLE
Data dictionary to logical model parser

### PR DESCRIPTION
**Pull request by ibacher**
This PR adds some very clunky named classes and a parser that reads the data dictionary, does some standarizations, and returns a usable version of the model.

Usage:

```python
parser = LogicalModelParser("path/to/test_dd.xlsx")
parser.parse_logical_model()

# this now holds a "logical" representation of the data elements in the data dictionary
parser.logical_model

# we also have
parser.dak_name
parser.cover_info # less useful, but if you want something from the cover page of the dd file, it should be here
```

Using the Logical Model:

```python
# ensure you've parsed things first
logical_model = parser.logical_model

logical_model.data_element_records # dict keyed from the data element's code to the DataElementRecord instance for this data element
logical_model.data_element_records_by_name # dict keyed from the data element's label to a list of DataElementRecords with that name
logical_model.invariants # dict keyed by the invariant code to the actual invariant

# e.g.
data_element = logical_model.data_element_records["HIV.C.DE8"] # contact with (suspected) exposure to HIV
data_element.activity_id # HIV.C3 Capture or update client history
data_element.data_element_id # HIV.C.DE8
data_element.data_element_label # Contact with and (suspected) exposure to HIV
data_element.description # The client had suspected or known exposure to HIV
data_element.multiple_choice_type # None
data_element.data_type # Boolean
data_element.input_options # True/False
data_element.quantity_subtype # None
data_element.calculation # None
data_element.required # RequiredType.OPTIONAL
data_element.condition_expression # None
data_element.decision_support_tables # None
data_element.aggregate_indicators # None
```

NB I haven't really tested the `generate_terminology_resources()` and `generate_logical_models()` functions, but they're probably less useful

### CODE REVIEW

File: who_l3_smart_tools/core/models/logical_model.py
Comment (xkmato) [inline who_l3_smart_tools/core/models/logical_model.py:1]: I love this approach. This is much easier to read.

File: who_l3_smart_tools/utils/__init__.py [hunk 1]
```diff
@@ -1,15 +1,4 @@
...
-    return "".join(
-        [
-            s.lower() if i == 0 else s.capitalize()
-            for i, s in enumerate(split_re.split(s))
-        ]
-    )
+# ruff: noqa: F403
+from .counter import *
+from .cql_helpers import *
+from .str import *
```
Comment (xkmato) [inline who_l3_smart_tools/utils/__init__.py:4]: These can become absolute imports since we decided to go with that approach.

**Review by xkmato — APPROVED** [review]
I don't know how I did not see this PR before. I love this approach. It makes everything easier to understand even for someone who does not have much domain knowledge. 

I think it makes sense to take this parsers approach everywhere.

File: who_l3_smart_tools/utils/__init__.py (hunk 1 above)
Comment (ibacher) [reply who_l3_smart_tools/utils/__init__.py:4]: Yeah, either way works. I sort of figured this is the one case where relative imports are actually less confusing about what's happening, but happy to change.

**Pull request merged**